src/foam/graphics/TreeGraph.js

```diff
@@ -263,33 +267,72 @@
              slot.set(newValue);
              return false;
            }
-           slot.set((14*slot.get() + newValue)/15);
+           slot.set((2*slot.get() + newValue)/3);
            return true;
          }
        ],
 
        listeners: [
-         {
-           name: 'doLayout',
-           isFramed: true,
-           documentation: 'Animate layout until positions stabilize',
-           code: function() {
-             var needsLayout = false;
-             // Scale and translate the view to fit in the available window
-             var gw = this.graph.width-110;
-             var w  = this.maxRight - this.maxLeft + 55;
-             if ( w > gw ) {
-               var scaleX = Math.min(1, gw / w);
-               needsLayout = this.convergeTo(this.scaleX$, scaleX) || needsLayout;
-             }
-
-             var x = (-this.maxLeft+25)/w * gw + 55;
-             needsLayout = this.convergeTo(this.x$, x) || needsLayout;
-             if ( this.layout() || needsLayout ) this.doLayout();
-             this.graph.invalidate();
-           }
-         }
+        {
+          name: 'doLayout',
+          isFramed: true,
+          documentation: 'Animate layout until positions stabilize',
+          code: function() {
+            var needsLayout = false;
+            // Scale and translate the view to fit in the available window
+            var gw = this.graph.width-110;
+            var w  = this.maxRight - this.maxLeft + 55;
+
+            var x = (-this.maxLeft+25)/w * gw + 55;
+            needsLayout = this.convergeTo(this.x$, x) || needsLayout;
+            
+            if ( this.layout() || needsLayout ) {
+              this.doLayout();
+            }
+            else {
+              this.graph.updateCWidth();
+            }
+          
+            this.graph.invalidate();
+          }
+        }
        ]
      }
+   ],
+
+   listeners: [
+     {
+       name: 'updateCWidth',
+       isFramed: true,
+       code: function() {
+        const maxes = {
+          maxLeft: 0,
+          maxRight: 0
+        };
+          
+        function traverseAndCompare(root) {
+          if ( root.maxLeft < maxes.maxLeft ) maxes.maxLeft = root.maxLeft;
+          if ( root.maxRight > maxes.maxRight ) maxes.maxRight = root.maxRight;
+
+          for ( var i = 0; i < root.children.length; i++ ){
+            traverseAndCompare(root.children[i], root.x);
+          }
+        }
+
+        traverseAndCompare(this.root);
+
+        // needed to use the adjustments to width in order to account for proper fitting on the screen
+        // since by default with out it, the leftmost node and right most node get cutoff by half
+        // padding adjustments are there to proper spacing and also for the edge connectors to 
+        // fully render
+        var width = Math.abs(maxes.maxLeft - maxes.maxRight) + this.nodeWidth + this.padding * 4;
```

Comment:
It feels like there's something deeper we don't really understand when it comes to the sizing here. Adding `nodeWidth` and `padding * 4` feels super random. `padding * 2` would make some sense because it's like you're adding left padding and right padding to the width. Why does adding a `nodeWidth` to the width of the canvas help?

Also, when we were working on the home button, wasn't the `x` value the center of the node rather than the far left side. Is this a detail possibly being overlooked? Or does maxLeft/maxRight already handle that?

Reply:
So without the adjustments, the canvas cuts off halfway on the leftmost node and the right most node. Thats where the intution of adding a nodeWidth helps to adjust for that part because its technically adding in two node halves in.